embed/etcd.go

```diff
@@ -400,6 +408,29 @@ func (e *Etcd) serve() (err error) {
 			e.errHandler(s.serve(e.Server, &e.cfg.ClientTLSInfo, h, e.errHandler))
 		}(sctx)
 	}
+
+	if len(e.cfg.ListenMetricsUrls) > 0 {
+		// TODO: maybe etcdhttp.MetricsPath or get the path from the user-provided URL
+		metricsMux := http.NewServeMux()
+		metricsMux.Handle("/metrics", prometheus.Handler())
+
+		for _, murl := range e.cfg.ListenMetricsUrls {
+			if murl.Scheme == "https" {
```

Comment:
I think it should respect http/https/unix/unixs like client/peer listeners. There are probably both use cases where the client TLS configuration is desirable and where plaintext is what's needed (for example, people asking for special plaintext /health listeners). Configuring based on the url scheme would handle these cases while also being consistent with the way other listen url flags work.